cmd/oras/internal/display/status/interface.go

```diff
@@ -30,3 +30,10 @@ type PushHandler interface {
 
 // AttachHandler handles text status output for attach command.
 type AttachHandler PushHandler
+
+type TagHandler interface {
+	// OnTagged is called when each tagging operation is done.
+	OnTagged(tag string) error
+	// PreTagging is called before tagging.
+	PreTagging(reference string)
+}
```

Comment:
General comment about these display handlers, they are very specific to the command and IMO should be part of the command rather than the status/printing. They are like tasks for the command or steps.

Reply:
Any good idea on the refactoring? This is only interface and will be used by handler factory (see handlers in `display` package) and command codes.

Reply:
I don't think they need to be refactored, it just seesm like they shouldn't be under `display`. I'll look at it more eventually, but I like to at least throw out these ideas for consideration.